activerecord/lib/active_record/store.rb

```diff
@@ -93,6 +93,30 @@ module ActiveRecord
   #       super.to_i
   #     end
   #   end
+  #
+  # The keys used to access the store can be transformed by the
+  # <tt>read_store_attribute</tt> and <tt>write_store_attribute</tt> private
```

Comment:
If they are private we shouldn't be documenting them.

There is an exception for documenting private API (inside the file) where RDoc won't export those comments to the documentation, as it will aid the reader who might be working on the implementation. As long as it doesn't end up on `api.rubyonrails.org`.

Reply:
They are marked as `# :doc`, which is a decision made prior to this commit. That exposed them to the API docs as methods with documented signatures without explanation.

Reply:
These methods were once marked at `protected`, but were changed to `private` as part of https://github.com/rails/rails/commit/5b14129d8d4ad302b4e11df6bd5c7891b75f393c#diff-42bb988889e5077d4a324ae830b7b0fceec880f40b8eb353a35d03f4699cbdb7 (authored by @amatsuda). The addition of the `# :doc:` comment was added at that time.

Reply:
Thanks for investigating! I would like to hear from Akira if the intention was to make them public API at the time.

But since they are already public, this seems fine. I was also just point that out for reference, but good catch!

Reply:
@zzak in response to https://github.com/rails/rails/pull/50304 being merged, I've changed this diff to promote both methods to `public`, and to remove the `:doc:` comments.

Reply:
> I would like to hear from Akira if the intention was to make them public API at the time.

No, I just wanted to narrow all these methods' scope without introducing any documentation change. So I'm almost sure that most of such methods (private methods with `:doc:`) at that time were not meant to be "public APIs", but just tagged so for this historical reason (the original author of the method didn't know what `privated` and `protected` really mean in Ruby, and just misused `protected`). I'm sorry I probably should have reviewed all such methods right after this change if we want to expose each them on the RDoc of not.

So, speaking of these two methods in this PR, I'm against making them public. So far as the purpose is just for documentation, you could use `:doc:` directive instead. Please propose to make them public in Ruby level only when you hit any actual use case that you need to call them publicly in your Ruby program.

Reply:
Thank you for sharing that history @amatsuda.

The original version of this change kept them `private`, but added method-level documentation. I'll revert to that version.

Reply:
> The original version of this change kept them private, but added method-level documentation.

Thank you! I'm happy with the change. This is exactly what I meant (haven't looked into the documentation details though. I'll leave it to @zzak).

Reply:
Thanks for the review and extra context. :bow:

I feel weird documenting private methods that are intended to be monkey-patched. That to me feels like an incomplete API. Is this something we've done before?

Reply:
@zzak [ActiveJob::Serializers::ObjectSerializer](https://api.rubyonrails.org/classes/ActiveJob/Serializers/ObjectSerializer.html) has class- and method-level documentation for a private [#klass](@zzak [ActiveJob::Serializers::ObjectSerializer](https://api.rubyonrails.org/classes/ActiveJob/Serializers/ObjectSerializer.html)) method.

For what it's worth, I think _that_ method should be renamed to `#object_class` and be made public 😈. Having said that, it's prior art.